images/metering-ansible-operator/roles/meteringconfig/library/k8s_event.py

```diff
@@ -0,0 +1,328 @@
+    type: dict
+  state:
+    description:
+    - Determines if an object should be created, patched, or deleted. When set to C(present), an object will be
+      created, if it does not already exist. If set to C(absent), an existing object will be deleted. If set to
+      C(present), an existing object will be patched, if its attributes differ from those specified using
+      module options and I(resource_definition).
+    default: present
+    choices:
+    - present
+    - absent
+  force:
+    description:
+    - If set to C(True), and I(state) is C(present), an existing object will be replaced.
+    default: false
+    type: bool
+  merge_type:
+    description:
+    - Whether to override the default patch merge approach with a specific type. By default, the strategic
+      merge will typically be used.
+    - For example, Custom Resource Definitions typically aren't updatable by the usual strategic merge. You may
+      want to use C(merge) if you see "strategic merge patch format is not supported"
+    - See U(https://kubernetes.io/docs/tasks/run-application/update-api-object-kubectl-patch/#use-a-json-merge-patch-to-update-a-deployment)
+    - Requires openshift >= 0.6.2
+    - If more than one merge_type is given, the merge_types will be tried in order
+    - If openshift >= 0.6.2, this defaults to C(['strategic-merge', 'merge']), which is ideal for using the same parameters
+      on resource kinds that combine Custom Resources and built-in resources. For openshift < 0.6.2, the default
+      is simply C(strategic-merge).
+    choices:
+    - json
+    - merge
+    - strategic-merge
+    type: list
+  name:
+    description:
+      - Use to specify a Event object name.
+    required: true
+    type: str
+  namespace:
+    description:
+      - Use to specify a Event object namespace.
+    required: true
+    type: str
+  message:
+    description:
+      - Status for Operation
+    required: true
+    type: str
+  reason:
+    description:
+      - Reason for the transition into the objects current status
+    required: true
+    type: str
+  reportingComponent:
+    description:
+      - Component responsible for event
+    required: true
+    type: str
+  type:
+    description:
+      - Specifies the type of Event to create.
+    choices:
+      - NodePort
+      - ClusterIP
+      - LoadBalancer
+      - ExternalName
+  source:
+    description:
+      - Component for reporting this Event
+    required: true
+    type: string
+  involvedObject:
+    description: ObjectReference
+      - Object event is reporting on. ApiVersion, kind, name and namespace are of the involvedObject.
+    - apiVersion
+      required: true
+      type: string
+    - kind
+      required: true
+      type: string
+    - name
+      required: true
+      type: string
+    - namespace
+      required: true
+      type: string
+
+requirements:
+  - python >= 2.7
+  - openshift >= 0.6.2
+"""
+
+EXAMPLES = """
+
+- name: Create Kubernetes Event
```

Comment:
Maybe generally for this example try to reproduce a real event that's generated in a real controller. A lot of the example values here aren't representative of a properly defined Event. Reason is a single word, CamelCase, message is often a useful description. Right now this has mostly place-holders so it's hard to figure out the proper conventions. Just do a `kubectl get events --all-namespaces -o yaml` and find one that seems like a good basis to try to recreate using the module.